src/Compilers/CSharp/Portable/FlowAnalysis/FlowAnalysisPass.cs

```diff
@@ -78,6 +80,25 @@ internal class FlowAnalysisPass
             return block;
         }
 
+        private static BoundBlock AppendImplicitReturn(BoundBlock body, bool originalBodyNested, MethodSymbol method, CSharpSyntaxNode syntax)
+        {
+            if (originalBodyNested)
```

Comment:
WasCompilerGenerated will be set in the case where there is a constructor initializer but no original body as well. Passing an explicit flag seemed less fragile.